debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/events/LogMinerEventRow.java

```diff
@@ -0,0 +1,210 @@
+ */
+package io.debezium.connector.oracle.logminer.events;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.connector.oracle.Scn;
+import io.debezium.relational.TableId;
+import io.debezium.util.HexConverter;
+import io.debezium.util.Strings;
+
+/**
+ * A simple wrapper around a {@link ResultSet} for a given row.
+ *
+ * @author Chris Cranford
+ */
+public class LogMinerEventRow {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LogMinerEventRow.class);
+
+    private static final Calendar UTC_CALENDAR = Calendar.getInstance(TimeZone.getTimeZone(ZoneOffset.UTC));
+
+    private static final int SCN = 1;
+    private static final int SQL_REDO = 2;
+    private static final int OPERATION_CODE = 3;
+    private static final int CHANGE_TIME = 4;
+    private static final int TX_ID = 5;
+    private static final int CSF = 6;
+    private static final int TABLE_NAME = 7;
+    private static final int TABLESPACE_NAME = 8;
+    private static final int OPERATION = 9;
+    private static final int USERNAME = 10;
+    private static final int ROW_ID = 11;
+    private static final int ROLLBACK_FLAG = 12;
+    private static final int RS_ID = 13;
+    private static final int HASH = 14;
+
+    private Scn scn;
+    private TableId tableId;
+    private String tableName;
+    private String tablespaceName;
+    private EventType eventType;
+    private Instant changeTime;
+    private String transactionId;
+    private String operation;
+    private String userName;
+    private String rowId;
+    private boolean rollbackFlag;
+    private String rsId;
+    private long hash;
+    private String redoSql;
+
+    public Scn getScn() {
+        return scn;
+    }
+
+    public TableId getTableId() {
+        return tableId;
+    }
+
+    public String getTableName() {
+        return tableName;
+    }
+
+    public String getTablespaceName() {
+        return tablespaceName;
+    }
+
+    public EventType getEventType() {
+        return eventType;
+    }
+
+    public Instant getChangeTime() {
+        return changeTime;
+    }
+
+    public String getTransactionId() {
+        return transactionId;
+    }
+
+    public String getOperation() {
+        return operation;
+    }
+
+    public String getUserName() {
+        return userName;
+    }
+
+    public String getRowId() {
+        return rowId;
+    }
+
+    public boolean isRollbackFlag() {
+        return rollbackFlag;
+    }
+
+    public String getRsId() {
+        return rsId;
+    }
+
+    public long getHash() {
+        return hash;
+    }
+
+    public String getRedoSql() {
+        return redoSql;
+    }
+
+    public LogMinerEventRow from(ResultSet resultSet, String catalogName, boolean isTxIdRawValue) throws SQLException {
```

Comment:
Would you prefer this static factory to always return a new instance as well?  I was attempting to minimize the number of objects created on each iteration of the `ResultSet` but I agree that with how I had to approach using this in the test, it felt clunky for sure.

Reply:
Well, yes, I think it can internally return the same instance but it has to be clrealy documented. So maybe even the name of method should reflect it somehow so definitely noething like `createNew`